### PR TITLE
Add iframe bridge documentation

### PR DESCRIPTION
**Pull request by manuelblum**
screencapture-localhost-3300-docs-preview-iframebridge-2024-11-22-12_49_37

### CODE REVIEW

File: packages/site/cms-site/src/iframebridge/IFrameMessage.ts [hunk 1]
```diff
@@ -69,34 +79,50 @@ export enum AdminMessageType {
...
 }
 
+/**
+ * The `IAdminHoverComponentMessage` is sent from the admin to the site, when a component is hovered in the admin interface.
+ */
 export interface IAdminHoverComponentMessage {
     cometType: AdminMessageType.HoverComponent;
     data: {
-        adminRoute: string;
+        adminRoute: string | null;
```
Comment (manuelblum) [inline packages/site/cms-site/src/iframebridge/IFrameMessage.ts:119]: unsure about this change. But iFrameMessages are the contract between admin and site. those types should be the same.

File: docs/docs/preview/iframebridge/IFrameBridge.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
+---
+title: IFrameBridge
+sidebar_position: 8
+---
+
+The Block Preview allows you to preview your blocks in the context of the Comet Admin. The Block Preview is an iframe that is embedded in the Comet Admin. Comet Admin is intended to be a headless CMS, but to offer the capability to render a preview of the content, the block-preview got introduced.
+
+![IFrame](images/iframe.png)
+
+The `IFrameBridge` is a communication bridge between the Comet Admin and the Block Preview implemented in the site's client. It allows you to send messages from the Comet Admin to the Block Preview and vice versa. In the admin client the `IFrameBridge` is already implemented and can be used out of the box and is already sending messages to the iframe site client's block preview. In the site client you can use the IFrameBridge from `@comet/cms-site` to receive/send messages from/to the Comet Admin.
```
Comment (johnnyomair) [inline docs/docs/preview/iframebridge/IFrameBridge.md:10]: Regarding "site's client": Isn't site and client the same?

File: docs/docs/preview/iframebridge/IFrameBridge.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+## IFrameMessages
+
+There are several messages that can be sent between the Comet Admin and the Block Preview. All of them are handled or are available in a higher level api which is provided in the `useIFrameBridge` hook.
+
+| Property                       | Sender | Description                                                                                                                                                                                                   | available                                  |
```
Comment (johnnyomair) [inline docs/docs/preview/iframebridge/IFrameBridge.md:16]: Out of scope: Maybe this table could be generated from source somehow.

File: docs/docs/preview/iframebridge/IFrameBridge.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+| `IAdminHoverComponentMessage`  | Admin  | The `IAdminHoverComponentMessage` is sent from the admin to the site, when a component is hovered in the admin interface.                                                                                     | `iFrameBridge.sendHoverComponent("id")`    |
+| `IAdminContentScopeMessage`    | Admin  | The `IAdminContentScopeMessage` is sent from the admin to the site, when the content scope is changed in the admin interface.                                                                                 | `iFrameBridge.sendContentScope(scope)`     |
+
+## block-preview
+
+To make the Block Preview embeddable in the Comet Admin, the client must provide an entry point where the client application renders the block preview and implements the `IFrameBridge` for communication. In a typical comet application the block preview is available at `site-domain.com/block-preview/page`
+
+To Implement the `IFrameBridge` in any site client, you can use `IFrameBridgeProvider` and `useIFrameBridge` from `@comet/cms-site`.
```
Comment (johnnyomair) [inline docs/docs/preview/iframebridge/IFrameBridge.md:33]: I believe you'd use the higher level `BlockPreviewProvider`

File: packages/site/cms-site/src/iframebridge/IFrameBridge.tsx [hunk 5]
```diff
@@ -32,7 +32,7 @@ export interface IFrameBridgeContext {
     block?: any;
     showOnlyVisible: boolean;
     selectedAdminRoute?: string;
-    hoveredAdminRoute?: string;
+    hoveredAdminRoute?: string | null;
```
Comment (johnnyomair) [inline packages/site/cms-site/src/iframebridge/IFrameBridge.tsx:35]: Why is this change necessary?

File: packages/site/cms-site/src/iframebridge/IFrameMessage.ts (hunk 1 above)
Comment (johnnyomair) [reply packages/site/cms-site/src/iframebridge/IFrameMessage.ts:119]: I'm not sure if the adminRoute can ever be null, so maybe it would be better to remove it in the admin?

Comment (manuelblum) [reply packages/site/cms-site/src/iframebridge/IFrameMessage.ts:119]: i think it can be null, when something gets "unhovered". I removed it first also in the admin, but then i thought typing adjustment would be better to add the null in the client.

File: docs/docs/preview/iframebridge/IFrameBridge.md (hunk 4 above)
Comment (manuelblum) [reply docs/docs/preview/iframebridge/IFrameBridge.md:33]: I was not aware of the BlockPreviewProvider, but i don't think that the BlockPreviewProvider is here the way to go. 

The only thing BlockPreviewProvider is doing, is storing a state of the previewType and if Preview Skeletons should be shown.

Screenshot 2024-11-29 at 08 50 00

If you want to communicate between admin and site, one has to use the IFrameBridge.

File: docs/docs/preview/iframebridge/IFrameBridge.md (hunk 2 above)
Comment (manuelblum) [reply docs/docs/preview/iframebridge/IFrameBridge.md:10]: agree, i though it would be a little bit clearer when haveing both words, but we can reduce it to client.

File: docs/docs/preview/iframebridge/IFrameBridge.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+| `IFrameOpenLinkMessage`        | Site   | @deprecated                                                                                                                                                                                                   |
+| `IFrameLocationMessage`        | Site   | @deprecated                                                                                                                                                                                                   |
+| `IAdminBlockMessage`           | Admin  | The `IAdminBlockMessage` is sent from the admin to the site, which block should be displayed in the iFrame.                                                                                                   | `iFrameBridge.sendBlockState()`            |
+| `IAdminShowOnlyVisibleMessage` | Admin  | The `IAdminBlockMessage` is sent from the admin to the site, which block should be displayed in the iFrame.                                                                                                   | `iFrameBridge.sendShowOnlyVisible(true)`   |
```
Comment (thomasdax98) [inline docs/docs/preview/iframebridge/IFrameBridge.md:24]: This looks like a copy mistake:

```suggestion
| `IAdminShowOnlyVisibleMessage` | Admin  | The `IAdminShowOnlyVisibleMessage` is sent from the admin to the site, indicating if hidden blocks should be rendered or not.                                                                                                   | `iFrameBridge.sendShowOnlyVisible(true)`   |
```

**Review by thomasdax98 — APPROVED** [review]

File: packages/site/cms-site/src/iframebridge/IFrameMessage.ts (hunk 1 above)
Comment (johnnyomair) [reply packages/site/cms-site/src/iframebridge/IFrameMessage.ts:119]: You're right, this was incorrect in the site, thanks!
